smart-contract-verifier/smart-contract-verifier/src/vyper/artifacts.rs

```diff
@@ -0,0 +1,295 @@
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CompilerInput {
+    pub language: String,
+    pub sources: ethers_solc::artifacts::Sources,
+    #[serde(default)]
+    pub interfaces: Interfaces,
+    #[serde(default)]
+    pub settings: Settings,
+}
+
+impl verifier::CompilerInput for CompilerInput {
+    fn modify(mut self) -> Self {
+        self.sources.iter_mut().for_each(|(_file, source)| {
+            let mut modified_content = source.content.as_ref().clone();
+            modified_content.push(' ');
+            source.content = std::sync::Arc::new(modified_content);
+        });
+        self
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Settings {
+    #[serde(
+        default,
+        with = "serde_helpers::display_from_str_opt",
+        skip_serializing_if = "Option::is_none"
+    )]
+    pub evm_version: Option<ethers_solc::EvmVersion>,
+    /// Indicates whether or not optimizations are turned on.
+    /// This is true by default.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub optimize: Option<bool>,
+    /// Indicates whether or not the bytecode should include Vyper's signature.
+    /// This is true by default.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub bytecode_metadata: Option<bool>,
+    /// This field can be used to select desired outputs based
+    /// on file and contract names.
+    /// If this field is omitted, then the compiler loads and does type
+    /// checking, but will not generate any outputs apart from errors.
+    #[serde(default)]
+    pub output_selection: FileOutputSelection,
+}
+
+impl Default for Settings {
+    fn default() -> Self {
+        Self {
+            output_selection: OutputSelection::default_file_output_selection(),
+            evm_version: None,
+            optimize: None,
+            bytecode_metadata: None,
+        }
+    }
+}
+
+mod interfaces {
+    use serde::{Deserialize, Serialize};
+
+    #[derive(Clone, Debug, Serialize, Deserialize, Eq, PartialEq)]
+    #[serde(rename_all = "camelCase")]
+    pub struct AbiSource {
+        pub abi: serde_json::Value,
+    }
+
+    #[derive(Clone, Debug, Serialize, Deserialize, Eq, PartialEq)]
+    #[serde(rename_all = "camelCase")]
+    pub struct ContractTypesSource {
+        pub contract_types: serde_json::Value,
+    }
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, Eq, PartialEq)]
+#[serde(untagged)]
+pub enum Interface {
+    Vyper(ethers_solc::artifacts::Source),
+    Abi(interfaces::AbiSource),
+    ContractTypes(interfaces::ContractTypesSource),
+}
+
+impl Interface {
+    pub fn try_new(path: &Path, content: String) -> Result<Self, anyhow::Error> {
+        // Adapted from: https://github.com/vyperlang/vyper/blob/v0.3.9/vyper/cli/vyper_compile.py#L221
+        if path.extension() == Some(std::ffi::OsStr::new("json")) {
```

Comment:
hm, just note that if file has name `.json` then path.extensions() will be an empty string, should we care about it?

Reply:
I believe, no. Because in that case that file could not be imported properly anyway